### PR TITLE
#UIC-849: Map flow issue

### PR DESCRIPTION
**Pull request by avantika-karwasara**
branch : dev/map server url flow
- when map url is empty than Run query button will be disable

### CODE REVIEW

File: superset/assets/src/explore/controls.jsx [hunk 1]
```diff
@@ -1802,6 +1802,44 @@ export const controls = {
...
+              props.options.value = props.options.default;
+            } else{
+              props.options.value = props.options.lastValue;
+            }
+          }
+        }
+      }
+      props.textChange = (e) => {
+        if(state.controls.hasOwnProperty('labels_outside')){
+          if(state.controls.labels_outside.value){
```
Comment (jitendra-kumawat) [inline superset/assets/src/explore/controls.jsx]: do we don't need to reset flags on each call

File: superset/assets/src/explore/controls.jsx [hunk 2]
```diff
@@ -1802,6 +1802,44 @@ export const controls = {
...
+            } else{
+              props.options.value = props.options.arcGISValue;
+            }
+          }else {
+            if(!props.options.isLastValueChange){
+              props.options.value = props.options.default;
+            } else{
+              props.options.value = props.options.lastValue;
+            }
+          }
```
Comment (jitendra-kumawat) [inline superset/assets/src/explore/controls.jsx]: can we write this function with less if else logic

File: superset/assets/src/explore/components/controls/TextControl.jsx [hunk 3]
```diff
@@ -51,8 +51,21 @@ export default class TextControl extends React.Component {
...
   }
+
+  componentWillReceiveProps(nextProps){
+    if(nextProps.value !== this.props.value){
+      if(typeof(this.props.onChange) === 'function'){
+        this.props.onChange(nextProps.value, [])
+      }
+    }
+  }
+
```
Comment (jitendra-kumawat) [inline superset/assets/src/explore/components/controls/TextControl.jsx]: can we use componentDidUpdate instead componentWillReceiveProps as it is unsafe as per
https://reactjs.org/docs/react-component.html

File: superset/assets/src/explore/controls.jsx (hunk 1 above)
Comment (avantika-karwasara) [reply superset/assets/src/explore/controls.jsx]: done

File: superset/assets/src/explore/controls.jsx (hunk 2 above)
Comment (avantika-karwasara) [reply superset/assets/src/explore/controls.jsx]: done

**Pull request merged**